### PR TITLE
Regen historical states for `new-state-mgmt` compatibility

### PR DESCRIPTION
**Pull request by terencechain**
Note: this is not a run time breaking feature, this is just a tool to let user convert non new-state-mgmt compatible DB to new-state-mgmt compatible. There's little to non risk 

What the workflow looks like, if historical states got pruned due to miss usage of `new-state-mgmt`,  and the users want to use `new-state-mgmt` again, DB will prompt user whether he/she wants to regenerate historical states. If user enters `Y`, the node will start regenerating historical states. For me, the regenerate process took about 45 mins to generate 500k slots worth of historical states at 256 slots intervals.

Screen Shot 2020-03-31 at 3 13 22 PM

Change list:
* Rename `/db/kv/check_state.go` to `/db/kv/check_historical_state.go`
* Set last archived index back to 0 running without `new-state-mgmt` as all historical states get pruned.
* Add questionnaire to confirm (y/n) user wants to regenerate historical states
* Add conditions to ensure historical state generation process can pick up where its left off in the event of interruption

### CODE REVIEW

File: beacon-chain/db/kv/regen_historical_states.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	currentState := genesisState.Copy()
+	startSlot := genesisState.Slot()
+	lastArchivedIndex, err := kv.LastArchivedIndex(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Restore from last archived point if this process was previously interrupted.
+	if lastArchivedIndex > 0 {
+		states, err := kv.HighestSlotStatesBelow(ctx, (lastArchivedIndex-1)*slotsPerArchivedPoint+1)
```
Comment (rauljordan) [inline beacon-chain/db/kv/regen_historical_states.go:54]: Can we make this second argument into its own variable to understand why we do the -1 and the * +1?

File: beacon-chain/db/kv/regen_historical_states.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	transition "github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db/filters"
+	stateTrie "github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+func (kv *Store) regenHistoricalStates(ctx context.Context) error {
```
Comment (rauljordan) [inline beacon-chain/db/kv/regen_historical_states.go:19]: This function does too much, it should be refactored into smaller pieces perhaps within this same file

File: beacon-chain/db/kv/regen_historical_states.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		return nil, errors.New("state can't be nil")
+	}
+	if state.Slot() > slot {
+		err := fmt.Errorf("expected state.slot %d < slot %d", state.Slot(), slot)
+		return nil, err
+	}
+	if state.Slot() == slot {
+		return state, nil
+	}
+	for state.Slot() < slot {
```
Comment (rauljordan) [inline beacon-chain/db/kv/regen_historical_states.go:166]: This logic is pretty similar to the state transition function. To reduce code duplication, is there a way we can abstract this core logic ?

File: beacon-chain/db/kv/regen_historical_states.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/db/kv/regen_historical_states.go:54]: Yep. Done

**Pull request merged**